examples/PyTorch/README.md

```diff
@@ -0,0 +1,98 @@
+
+### Distributed
+
+To make the code distributed apply provided patch by running:
+
+```
+git apply distributed.patch
+```
+
+You might want to checkout commit 0c1654d6913f77f09c0505fb284d977d89c17c1a to make sure there are no conflicts before applying the patch.
+
+To learn how to easily adapt PyTorch code for distributed learning please examine the patch. 
+
+## Usage
+```
+# set accroding to your configuration
+export GLOO_SOCKET_IFNAME=eth0 
+```
+
+```
+CUDA_VISIBLE_DEVICES=0 python main.py --init-method tcp://127.0.0.1:20011 --rank 0 --world 2 --dataset lsun --dataroot <path to lsun dataset> --cuda
+CUDA_VISIBLE_DEVICES=1 python main.py --init-method tcp://127.0.0.1:20011 --rank 1 --world 2 --dataset lsun --dataroot <path to lsun dataset> --cuda
+
+For more details about the usage refer to the [original repo](https://github.com/pytorch/examples/tree/master/dcgan)
+```
+
+## Running without tensorhive
+
+First export envoirment variable on every node as shown in Usage section. After find a port that is available on every node for communication. Then run separate commands as above on very node. Make sure to start with rank 0. Set world size to number of nodes you want to run the code on. The training code and dataset needs to be avaiable on every node.
+
+## Running with tensorhive
+Head to "Task Overview" (1) and click on "CREATE TASKS FROM TEMPLATE". Choose PyTorch from the drop-down list. 
+
+![1](https://github.com/roscisz/TensorHive/tree/master/examples/PyTorch/img/1.png)
+
+Choose host and resource to run the task on for the first node. Under command specify path to your pyhon and script to wish to run (as shown). Specify init-method and world-size (number of nodes you wish to distribute the training between).
+
+![2](https://github.com/roscisz/TensorHive/tree/master/examples/PyTorch/img/2.png)
+
+Add as many tasks as resources you wish the code to run on using "ADD TASK" button. You can see that every parameter filled is copied to newly created tasks to save time. Adjust Hostnames and resources on the added tasks as needed. And fill in the rank parameter starting from 0.
+
+![3](https://github.com/roscisz/TensorHive/tree/master/examples/PyTorch/img/3.png)
+
+Add GLOO_SOCKET_IFNAME envoirmental variable. By filling in the "Parameter name" below the tasks and clicking on "ADD AS ENV VARIABLE TO ALL TASKS" button. If you check the "Static" checkbox next to it you will be able to fill it only for one task and it will be copied to the rest of them. For most cases you will have the same connection configuration on all nodes so it can save you some effort. Fill the newly added env variable according to the configuration on your nodes. You can check the names of your internet interfaces by running `ifconfig` command on your nodes.
+
+Click "CREATE ALL TASKS" button in the right bottom corner to create the tasks.
+
+![4](https://github.com/roscisz/TensorHive/tree/master/examples/PyTorch/img/4.png)
```

Comment:
In img 4 there is a space between GLOO_SOCKET_IFNAME and =eth0. Would it work like that? Did you actually execute this?

Reply:
That was a typo (fixed). I was running slightly different task with correct command and a local interface